File: 100-Days-Series/Day 65/DiamandPatternWithNumber.py

# Function to pritn Diamand Pattern With Numbers.
def get_print_pattern(rows):
    # Upper Part
    for i in range(1,rows+1):
        for space in range(0,rows-i):
            print(end=" ")

        for j in range(0,(2*i-1)):
            print(i,end="")
        print()

    # Lower Part
    for i in range(rows-1,0,-1):
        for space in range(0,rows-i):
            print(end=" ")

        for j in range(0,(2*i-1)):
            print(i,end="")
        print()


# Let's Start Here
rows=int(input("Enter the number of rows : "))
get_print_pattern(rows)

File: 100-Days-Series/Day 65/DiamandPatternWithNumber.cpp
#include<iostream>
using namespace std;
// Function to print Diamand Pattern With Numbers.
void getPrintPattern(int rows){
    // Upper Part 
    for(int i=1;i<=rows;i++){
        for(int space=1;space<=rows-i;space++){
            cout<<" ";
        }
        for(int j=1;j<=(2*i-1);j++){
            cout<<i;
        }
        cout<<endl;
    }
    // Lower Part
    for(int i=rows-1;i>=1;i--){
        for(int space=1;space<=rows-i;space++){
            cout<<" ";
        }
        for(int j=1;j<=(2*i-1);j++){
            cout<<i;
        }
        cout<<endl;
    }
}
int main(){
    int rows;
    cout<<"\nEnter the number of rows : ";
    cin>>rows;
    getPrintPattern(rows);
}